include/deal.II/fe/fe_series.h

```diff
@@ -526,17 +603,16 @@ FESeries::Fourier<dim, spacedim>::save_transformation_matrices(
   // Store information about those resources which have been used to generate
   // the transformation matrices.
   // mode vector
-  unsigned int size = k_vectors.n_elements();
-  ar &         size;
+  ar &n_coefficients_per_direction;
 
   // finite element collection
-  size = fe_collection->size();
-  ar &size;
+  unsigned int size = fe_collection->size();
+  ar &         size;
   for (unsigned int i = 0; i < size; ++i)
     ar &(*fe_collection)[i].get_name();
 
   // quadrature collection
-  size = q_collection->size();
+  size = q_collection.size();
   ar &size;
   for (unsigned int i = 0; i < size; ++i)
     ar &(*q_collection)[i];
```

Comment:
```suggestion
    ar &(q_collection)[i];
```
In addition, the variable also appears in line 658, 663, 693, 696, 736, 741.